core/src/main/java/org/pentaho/platform/util/StringUtil.java

```diff
@@ -116,4 +118,31 @@ public static String getMapAsPrettyString( final Map map ) {
     MapUtils.debugPrint( new PrintStream( writtenBytes ), "Map", map );
     return writtenBytes.toString();
   }
+
+  /**
+   * Returns a map of String key-value pairs, in which each key and corresponding value has been trimmed of whitespace.
+   * @param map the {@link Map} of String key-value pairs to be trimmed
+   * @return a trimmed String {@link Map}
+   */
+  public static Map<String, String> trimStringMap( Map<String, String> map ) {
+    return map == null ? map : map.entrySet().stream().collect( Collectors.toMap( e -> StringUtils.trimToEmpty( e.getKey() ), e -> StringUtils.trimToEmpty( e.getValue() ) ) );
+  }
+
+  /**
+   * Returns a List of Strings in which each String has been trimmed of whitespace.
+   * @param list the {@link List} of Strings to be trimmed
+   * @return a trimmed {@link List} of Strings
+   */
+  public static List<String> trimStringList( List<String> list ) {
+    return list == null ? list : list.stream().map( StringUtils::trimToEmpty ).collect( Collectors.toList() );
+  }
+
+  /**
+   * Wrapper method around StringUtils.trim
+   * @param string the {@link String} to be trimmed
+   * @return the trimmed {@link String}
+   */
+  public static String trim( String string ) {
+    return StringUtils.trimToEmpty( string );
```

Comment:
I would use trim here, it's a bit misleading to have a class called StringUtil (very close to StringUtils) and have its trim method some something other than apache StringUtils' trim method. If we want the trimToEmpty functionality in pentaho's StringUtil, we could add a method with the same name.